PowerShellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingServiceUtilities.cs

```diff
@@ -345,5 +346,44 @@ private void AddEventHandlers()
             _runspace.StateChanged += _runspace_StateChanged;
             _runspace.AvailabilityChanged += _runspace_AvailabilityChanged;
         }
+
+        /// <summary>
+        /// Invokes given script on the provided PowerShell object after setting its runspace object.
+        /// </summary>
+        /// <param name="powerShell">This should usually be _currentPowerShell. Make sure to enclose uses of _currentPowerShell and this method
+        /// inside of a using. </param>
+        /// <param name="script">Script to invoke.</param>
+        /// <returns>Returns the result of the invoke</returns>
+        private Collection<PSObject> InvokeScript(PowerShell powerShell, string script)

```

Comment:
Let me share you the reason why I question about it, if you ever try to make such a common helper function, there are generally one minimal bar needs to consider: No(or at least minimize) implicit contract introduced to caller, you can not only think about the helper function as a shortcut of couple of lines of code.
So talk about your implementation, I would worry about your function would throw NRE from time to time just because there is an implicit contract, coz people can easily just pass in memeber _currentPowerShell object into your function without using statement to actually create the real object before calling it.